modelexpress_common/src/config.rs

```diff
@@ -249,31 +261,24 @@ where
 {
     let mut builder = Config::builder();
 
-    // Add configuration file if specified or if default exists
+    // Only load config file if explicitly provided
     if let Some(config_path) = &config_file {
-        if config_path.exists() {
-            builder = builder.add_source(File::from(config_path.clone()));
-        } else {
+        if !config_path.exists() {
             return Err(ConfigError::Message(format!(
                 "Configuration file not found: {}",
                 config_path.display()
             )));
         }
+        builder = builder.add_source(File::from(config_path.clone()));
+    } else if let Some(default_path) = discover_default_config() {
+        println!("Using default config: {}", default_path.display());
```

Comment:
just saw this now. we should probably use `info!` to be consistent with other loggings.